src/status_im/contexts/shell/share/wallet/view.cljs

```diff
@@ -0,0 +1,94 @@
+                                                  :content address}
+                               :item             {:default {:type "text"
+                                                            :content
+                                                            address}}
+                               :link-metadata    {:title share-title}}]}
+     {:title   share-title
+      :subject share-title
+      :message address})))
+
+(defn- open-preferences
+  [selected-networks]
+  (rf/dispatch [:show-bottom-sheet
+                {:theme :dark
+                 :shell? true
+                 :content
+                 (fn []
+                   [network-preferences/view
+                    {:blur?             true
+                     :selected-networks (set @selected-networks)
+                     :on-save           (fn [chain-ids]
+                                          (rf/dispatch [:hide-bottom-sheet])
+                                          (reset! selected-networks (map #(get utils/id->network %)
+                                                                         chain-ids)))}])}]))
+(defn wallet-qr-code-item
+  [account width index]
+  (let [selected-networks (reagent/atom [:ethereum :optimism :arbitrum])
+        wallet-type       (reagent/atom :wallet-legacy)]
+    (fn []
+      (let [share-title         (str (:name account) " " (i18n/label :t/address))
+            qr-url              (utils/get-wallet-qr {:wallet-type       @wallet-type
+                                                      :selected-networks @selected-networks
+                                                      :address           (:address account)})
+            qr-media-server-uri (image-server/get-qr-image-uri-for-any-url
+                                 {:url         qr-url
+                                  :port        (rf/sub [:mediaserver/port])
+                                  :qr-size     qr-size
+                                  :error-level :highest})]
+        [rn/view {:style {:width width :margin-left (if (zero? index) 0 -30)}}
+         [rn/view {:style style/qr-code-container} 
+          [quo/share-qr-code
+           {:type                @wallet-type
+            :qr-image-uri        qr-media-server-uri
+            :qr-data             qr-url
+            :networks            @selected-networks
+            :on-share-press      #(share-action qr-url share-title)
+            :profile-picture     nil
+            :unblur-on-android?  true
+            :full-name           (:name account)
+            :customization-color (:color account)
+            :emoji               (:emoji account)
+            :on-multichain-press #(reset! wallet-type :wallet-multichain)
+            :on-legacy-press     #(reset! wallet-type :wallet-legacy)
+            :on-settings-press   #(open-preferences selected-networks)}]]]
+      ))))
+
+(defn wallet-tab
+  []
+  (let [accounts (rf/sub [:wallet/accounts])
```

Comment:
`rf/sub [:wallet/accounts])` also includes watched addresses. Are watched addresses intended to be displayed?

Reply:
Watched addresses offer a dual purpose: they allow watching any address, as well as monitoring your own personal accounts without divulging keys. Consequently, both "share" and "receive" interfaces will display these accounts. Pedro and Mario are planning to introduce a new component state for the "Share QR Code", which will include support for watched accounts in the near future. @pedro-et @mariocnovoa

Reply:
thanks @xAlisher